docs/3_development/4_documentation.md

```diff
@@ -29,14 +29,64 @@ Don't forget to update repository settings as described in [Setup GitHub Reposit
-which make it hard to view changes in the browser. Deploying to [Render](https://render.com/) requires
-a one-time setup for which, if you don't have access to ChainSafe's Render account, you can ask your Department Head or DevOps team.
+The result of the Docusaurus build step is a static website that can be deployed to
+any of the popular hosting services including GitHub Pages, CloudFlare or IPFS.
+
+#### Deploy to CloudFlare Pages
+
+One way to achieve this is by deploying their websites on CloudFlare Pages.
+
+CloudFlare Pages is a modern platform for building and deploying websites. It offers several benefits, including:
+
+- High-performance: CloudFlare Pages uses the same global network as CloudFlare's CDN, which means websites load quickly no matter where the visitor is located.
+- Easy to use: Setting up a website on CloudFlare Pages is simple. You can connect your GitHub repository and deploy your site with just a few clicks.
+- Security: CloudFlare Pages offers built-in security features such as HTTPS encryption and DDoS protection.
+
+While CloudFlare workers is a relatively inexpensive option for Continuous Deployment, it still has quotas and requires a monthly subscription. On the other hand, GitHub Actions is free for public repositories so it's a preferred way of deploying websites.
+
+Here's an example of how you can deploy your website using GitHub Actions:
+
+1. [Obtain access to "ChainSafeDev" CloudFlare account](https://github.com/ChainSafe/cloudflare-hosting-mgmt/blob/master/members.tf) if you don't have one already.
+2. Create new Pages project either via CloudFlare Dashboard or [wrangler](https://developers.cloudflare.com/workers/wrangler/install-and-update/)
+3. Create a workflow file in their GitHub repository (e.g. .github/workflows/deploy.yml).
+4. Add a step to the workflow that deploys the built site to CloudFlare Pages using the CloudFlare Pages API.
+5. Set new chainsafe.dev subdomain available inside the "ChainSafeDev" CloudFlare account.
+
+Here's a sample workflow file:
+
+```yaml
+name: CloudFlare Deploy
+on:
+  workflow_run:
+    workflows:
+      - CI
+    types:
+      - completed
+
+jobs:
+  deploy:
+      runs-on: ubuntu-latest
+      if: ${{ github.event.workflow_run.conclusion == 'success' }}
+      permissions:
+        contents: read
+        deployments: write
+      steps:
+          - uses: actions/checkout@v3
+          - uses: actions/setup-node@v3
+            with:
+              cache: yarn
+              node-version: '16'
+          - run: yarn install --frozen-lockfile
+          - run: yarn run build
+          - name: Publish to Cloudflare Pages
+            uses: cloudflare/pages-action@1
+            with:
+              apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+              accountId: 2238a825c5aca59233eab1f221f7aefb
+              projectName: <cloudflare project name>
+              directory: ./build
+              gitHubToken: ${{ secrets.GITHUB_TOKEN }}
+```
 
```

Comment:
```suggestion
:::tip

To find PR-specific deployment URL go to the `Actions` tab, select deployment workflow on the left and choose one corresponding your PR number. The URL will be located in the `deploy summary` section.

:::
```

Reply:
I feel like we should experiment more, if we give certain permissions to workflow, maybe it will be able to create and link deployment to PR?